eZ/Publish/Core/FieldType/Image/ImageThumbnailStrategy.php

```diff
@@ -43,12 +52,34 @@ public function getFieldTypeIdentifier(): string
 
     public function getThumbnail(Field $field, ?APIVersionInfo $versionInfo = null): ?Thumbnail
     {
-        /** @var \eZ\Publish\SPI\Variation\Values\ImageVariation $variation */
-        $variation = $this->variationHandler->getVariation(
-            $field,
-            $versionInfo ?? new VersionInfo(),
-            $this->variationName
-        );
+        try {
+            /** @var \eZ\Publish\SPI\Variation\Values\ImageVariation $variation */
+            $variation = $this->variationHandler->getVariation(
+                $field,
+                $versionInfo ?? new VersionInfo(),
+                $this->variationName
+            );
+        } catch (SourceImageNotFoundException $e) {
+            $this->logger->warning(
+                sprintf(
+                    'Source image for thumbnail could not be found %s',
+                    $e->getMessage()
+                )
+            );
+
+            return null;
+        } catch (Exception $e) {
+            $this->logger->warning(
+                sprintf(
+                    'Thumbnail could not be generated for %s field and %s variation. Error: %s',
+                    $field->fieldTypeIdentifier,
+                    $this->variationName,
+                    $e->getMessage()
```

Comment:
The message in both cases should contain all the information needed to identify the Content item which causes the issue.
* if `$versionInfo` is not null: log Content Id + Version No
* `$field->fieldTypeIdentifier` - missing only for the first message
* `$this->variationName`  - missing only for the first message